src/protocols/http/hpr_http_roaming_worker.erl

```diff
@@ -0,0 +1,258 @@
+%%% @copyright (C) 2022, <COMPANY>
+%%% @doc
+%%%
+%%% @end
+%%% Created : 21. Sep 2022 1:01 PM
+%%%-------------------------------------------------------------------
+-module(hpr_http_roaming_worker).
+-author("jonathanruttenberg").
+
+-behavior(gen_server).
+
+-include("hpr_http_roaming.hrl").
+
+%% ------------------------------------------------------------------
+%% API Function Exports
+%% ------------------------------------------------------------------
+-export([
+    start_link/1,
+    handle_packet/4
+]).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Exports
+%% ------------------------------------------------------------------
+-export([
+    init/1,
+    handle_call/3,
+    handle_cast/2,
+    handle_info/2,
+    terminate/2
+]).
+
+-define(SERVER, ?MODULE).
+-define(SEND_DATA, send_data).
+-define(SHUTDOWN, shutdown).
+-define(SHUTDOWN_TIMEOUT, timer:seconds(5)).
+
+-type address() :: binary().
+
+-record(state, {
+    net_id :: hpr_http_roaming:netid_num(),
+    address :: address(),
+    transaction_id :: integer(),
+    packets = [] :: list(hpr_http_roaming:packet()),
+
+    send_data_timer = 200 :: non_neg_integer(),
+    send_data_timer_ref :: undefined | reference(),
+    flow_type :: async | sync,
+    auth_header :: null | binary(),
+    protocol_version :: pv_1_0 | pv_1_1,
+
+    should_shutdown = false :: boolean(),
+    shutdown_timer_ref :: undefined | reference()
+}).
+
+%% ------------------------------------------------------------------
+%% API Function Definitions
+%% ------------------------------------------------------------------
+
+start_link(Args) ->
+    gen_server:start_link(?SERVER, Args, []).
+
+-spec handle_packet(
+    WorkerPid :: pid(),
+    PacketUp :: hpr_packet_up:packet(),
+    GatewayTime :: hpr_http_roaming:gateway_time(),
+    GatewayStream :: hpr_router_stream_manager:gateway_stream()
+) -> ok | {error, any()}.
+handle_packet(Pid, PacketUp, GatewayTime, GatewayStream) ->
+    lager:debug(
+        [
+            {pid, Pid},
+            {packet_up, PacketUp},
+            {gateway_time, GatewayTime},
+            {gateway_stream, GatewayStream}
+        ],
+        "handle_packet"
+    ),
+    gen_server:cast(Pid, {handle_packet, PacketUp, GatewayTime, GatewayStream}).
+
+%% ------------------------------------------------------------------
+%% gen_server Function Definitions
+%% ------------------------------------------------------------------
+init(Args) ->
+    #{
+        protocol := #http_protocol{
+            endpoint = Address,
+            flow_type = FlowType,
+            dedupe_timeout = DedupeTimeout,
+            auth_header = Auth,
+            protocol_version = ProtocolVersion
+        },
+        net_id := NetID
+    } = Args,
+    lager:debug("~p init with ~p", [?MODULE, Args]),
+    {ok, #state{
+        net_id = NetID,
+        address = <<Address/binary, <<"/">>/binary, <<"uplink">>/binary>>,
```

Comment:
Another setting we need to provide in the http protocol settings is a path.

Reply:
I don't understand why this is needed.

Reply:
The `address` expression can be `<<Address/binary, "/uplink">>` which is a little more clear.

Reply:
"path" has been added.